pkg/clustermanager/cluster_manager.go

```diff
@@ -177,7 +233,42 @@ func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, c
 }
 
 func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster, workloadCluster *types.Cluster, clusterSpec *cluster.Spec, provider providers.Provider) error {
-	if err := c.applyCluster(ctx, managementCluster, workloadCluster, clusterSpec, provider, true, types.WithNodeRef(), types.WithNodeHealthy()); err != nil {
+	cpContent, mdContent, err := provider.GenerateClusterApiSpecForUpgrade(ctx, managementCluster, workloadCluster, clusterSpec)
+	if err != nil {
+		return fmt.Errorf("error generating capi spec: %v", err)
+	}
+
+	if err = c.writeCapiSpecFile(clusterSpec.ObjectMeta.Name, templater.AppendYamlResources(cpContent, mdContent)); err != nil {
+		return err
+	}
+
+	err = c.Retrier.Retry(
+		func() error {
+			return c.clusterClient.ApplyKubeSpecFromBytesWithNamespace(ctx, managementCluster, cpContent, constants.EksaSystemNamespace)
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("error applying capi control plane spec: %v", err)
+	}
+
+	if clusterSpec.Spec.ExternalEtcdConfiguration != nil {
```

Comment:
I think these checks are already being done in this function [line 275](https://github.com/aws/eks-anywhere/pull/264/files#diff-2a3001a4bccb1fa62c0978f41ec1a89279d6e8cbc47e531f59c754b5df5715b6R275) onwards

Reply:
this is a confusing part for me too. so the original code actually check the etcd status twice. i thought we did it on purpose.

Reply:
chatted with rajashree offline, it seems double check etcd ready status is not necessary. gonna remove one